### PR TITLE
First fix attempt at the empty_action, to remove empty directories

### PR DESCRIPTION
**Pull request by mrguilima**
For manually testing it, do this within the _consistency pod_: 
 ```console
cd /consistency/cms_consistency/site_cmp3
source /consistency/cms_consistency/site_cmp3/empty-test.sh T2_US_Nebraska
```
and look at the output on `/var/cache/test/`.  Site `T2_US_Nebraska` is currently the only site with enough data left to trigger removal of empty diretories (its config file, <last>_config.yaml, has been relaxed to take existing data and used it).

It is currently set for 10 directories removed (-L 10), and those first 10 have already been removed during manual tests.  So the -L argument must be increased if any side effect is to be observed.  Default is None, and the original hardcoded value is 10000.

For enabling the change for real:
```console
  cd /consistency/cms_consistency/site_cmp3
  cp action/remove_empty_dirs_GL.py action/remove_empty_dirs.py
```

### CODE REVIEW

**Comment by mrguilima** [comment]
Some additional comments:

1) Not sure whether the script is safe regarding removing directories when they are not empty anymore.  E.g. since the script is based on data between 3-25 days old, in case a directory gets populated within the last two days, does the removal fail and contents survive?  I assume so, but this scenario has not been tested.

2) The `empty_action` configuration is at the moment based on some parameters from the `dark_action:` section:
```console
    dark_action:
      confirmation_window: 35
      max_age_last_run: 6
      max_fraction: 0.01
      min_age_first_run: 18
      min_runs: 3
```
see file: /var/cache/consistency-dump/T2_US_Nebraska_2024_08_17_19_34_config.yaml.  The `empty_action` probably shall eventually have its own configuration section.

3) Once "low volume" is further tested and validated, the parameter -L should be increased (original 10000) or dropped, in latter case all "confirmed empty dirs" (based on all data within 3-25 days window) will be removed.

File: actions/remove_empty_dirs.py [hunk 1]
```diff
@@ -7,11 +7,8 @@
 from rucio_consistency import CEConfiguration, Stats
 from rucio_consistency.xrootd import XRootDClient
 
-from run import CCRun
-from config import ActionConfiguration
-
 
-Version = "1.0"
+Version = "1.1"
```
Comment (dynamic-entropy) [inline actions/remove_empty_dirs.py:11]: Is this separate versioning of components required? 

I hope we can get rid of this in the future and keep a single app version.

File: actions/remove_empty_dirs.py [hunk 2]
```diff
@@ -323,8 +322,8 @@ def empty_action(storage_path, rse, out, lfn_converter, stats, stats_key, dry_ru
 
 storage_path, rse = args
 
-config = EmptyActionConfiguration(rse, opts["-c"])
-scanner_config = ScannerConfiguration(rse, opts["-c"])
+config  = ActionConfiguration(rse, opts["-c"], "dark")
```
Comment (dynamic-entropy) [inline actions/remove_empty_dirs.py:325]: Why is "dark" used as an `action` here?

**Review by dynamic-entropy — APPROVED** [review]
This has been running for quite some time. 
I will put this into a release for that reason.

I suggest discussing this in a meeting if required. 

FYI @eachristgr

File: actions/remove_empty_dirs.py (hunk 2 above)
Comment (dynamic-entropy) [reply actions/remove_empty_dirs.py:325]: Ah, found my answer in your comment above. 
```
    dark_action:
      confirmation_window: 35
      max_age_last_run: 6
      max_fraction: 0.01
      min_age_first_run: 18
      min_runs: 3
```

yes, we can add a new section for empty configuration.

**Comment by dynamic-entropy** [comment]
> Not sure whether the script is safe regarding removing directories when they are not empty anymore. E.g. since the script is based on data between 3-25 days old, in case a directory gets populated within the last two days, does the removal fail and contents survive? I assume so, but this scenario has not been tested.

@mrguilima Is the action not immediate after the scan?
I do not understand why its 3-25 days and not now-25 days.
The action should be done post-scan immediately.

**Pull request merged**